mono/mini/aot-compiler.c

```diff
@@ -10436,7 +10438,7 @@ mono_compile_assembly (MonoAssembly *ass, guint32 opts, const char *aot_options)
 	}
 
 #if defined(MONO_ARCH_GSHAREDVT_SUPPORTED)
-	if (acfg->aot_opts.llvm_only || mono_aot_mode_is_full (&acfg->aot_opts)) {
+	if (acfg->aot_opts.llvm_only || mono_aot_mode_is_full (&acfg->aot_opts) || mono_aot_mode_is_hybrid (&acfg->aot_opts)) {
```

Comment:
Would it make sense to bottleneck (mono_aot_mode_is_full (&acfg->aot_opts) || mono_aot_mode_is_hybrid (&acfg->aot_opts)) on a no_il_available() or gsharedvt_enabled() or something? I notice you use those checks in two places, and it is a little opaque to me looking through this code to understand why the choice was made between which of (full aot, llvm, hybrid) is checked for any particular if clause.